src/sidebar/components/ShareDialog/ImportAnnotations.tsx

```diff
@@ -135,11 +133,27 @@ function ImportAnnotations({
     );
   }
 
+  // In order to perform an import, we need:
+  //
+  //  1. A group to import into
+  //  2. A frame from which to get the document URI and metadata
+  //  3. Existing annotations for the group loaded so we can de-duplicate against
+  //     them
+  //  4. A file to import from and a selection of what to import
+  //     (`importAnnotations` will be falsey if this is not the case).
+  const importReady = Boolean(
+    store.focusedGroup() &&
+      store.mainFrame() &&
+      store.hasFetchedAnnotations() &&
+      !store.isFetchingAnnotations() &&
```

Comment:
Wouldn't `!store.isFetchingAnnotations()` cover the `store.hasFetchedAnnotations()` condition as well?

Reply:
When the application first starts up, it needs to fetch various pieces of information (eg. user profile, confirm the initial focused group ID, document URL) before it can start fetching annotations. During that time, no annotations have yet been fetched, but no annotation fetches are in progress either. When you switch groups later however, an annotation fetch can start right away.

Reply:
Some of these pieces of information (eg. user profile, group ID) are also included in the conditional here (eg. do we have a group and a frame). But I believe there are some other scenarios where there is additional info that is not included here, and also the prerequisites of fetching annotations could change over time, so it makes sense to explicitly check `hasFetchedAnnotations` here.